src/NServiceBus.Transport.RabbitMQ.Tests/When_consuming_messages.cs

```diff
@@ -58,11 +58,53 @@ public void Should_move_message_without_message_id_to_error_queue()
 
                 var result = channel.BasicGet(ErrorQueue, true);
 
-                Assert.False(messageWasReceived, "Message should not be processed processed successfully.");
+                Assert.False(messageWasReceived, "Message should not be processed successfully.");
                 Assert.NotNull(result, "Message should be considered poison and moved to the error queue.");
             }
         }
 
+        [Test]
+        public async Task Should_handle_retries_for_messages_without_headers()
```

Comment:
The problem should only manifest when using classic queues. It could be worth having a test that shows that it works with quorum queues as well without needing the fix.

Reply:
I've added this in a separate PR https://github.com/Particular/NServiceBus.RabbitMQ/pull/1224